packages/web-features-figuring/src/baseline/index.ts

```diff
@@ -0,0 +1,241 @@
+    : compatKeys;
+
+  const statuses = keys.map((key) => calculate(key, compat));
+  const support = collateSupport(statuses.map((status) => status.support));
+
+  const keystoneDate = findKeystoneDate(
+    statuses.flatMap((s) => [...s.support.values()]),
+  );
+  const { baseline, baseline_low_date, baseline_high_date } =
+    keystoneDateToStatus(keystoneDate);
+
+  return {
+    baseline,
+    baseline_low_date,
+    baseline_high_date,
+    support,
+    toJSON: function () {
+      return jsonify(this);
+    },
+  };
+}
+
+/**
+ * Compute the Baseline support ("high", "low" or false, dates, and releases)
+ * for a single compat key.
+ */
+function calculate(compatKey: string, compat: Compat): SupportStatus {
+  const f = feature(compatKey);
+  const s = support(f, browsers(compat));
+  const keystoneDate = findKeystoneDate([...s.values()]);
+
+  const { baseline, baseline_low_date, baseline_high_date } =
+    keystoneDateToStatus(keystoneDate);
+
+  return {
+    compatKey,
+    baseline,
+    baseline_low_date,
+    baseline_high_date,
+    support: s,
+    toJSON: function () {
+      return jsonify(this);
+    },
+  };
+}
+
+/**
+ * Given a compat key, get the key and any of its ancestor features.
+ *
+ * For example, given the key `"html.elements.a.href"`, return
+ * `["html.elements.a", "html.elements.a.href"]`.
+ */
+function withAncestors(compatKey: string, compat: Compat): string[] {
+  const items = compatKey.split(".");
+  const ancestors: string[] = [];
+
+  let current = items.shift();
+  while (items.length) {
+    current = `${current}.${items.shift()}`;
+
+    const data = compat.query(current);
+    if (typeof data === "object" && data !== null && "__compat" in data) {
+      ancestors.push(current);
+    }
+  }
+  return ancestors;
+}
+
+/**
+ * Collate several support summaries, taking the most-recent release for each browser across all of the summaries.
+ */
+function collateSupport(
+  supports: Map<Browser, Release | undefined>[],
+): Map<Browser, Release | undefined> {
+  const collated = new Map<Browser, (Release | undefined)[]>();
+
+  for (const support of supports) {
+    for (const [browser, release] of support) {
+      collated.set(browser, [...(collated.get(browser) ?? []), release]);
+    }
+  }
+
+  const support: Map<Browser, Release | undefined> = new Map();
+  for (const [browser, releases] of collated) {
+    if (releases.includes(undefined)) {
+      support.set(browser, undefined);
+    } else {
+      support.set(
+        browser,
+        releases
+          .sort((r1, r2) => (r1 as Release).compare(r2 as Release))
+          .at(-1),
+      );
+    }
+  }
+  return support;
+}
+
+/**
+ * Given several dates, find the most-recent date and determine the
+ * corresponding Baseline status and high and low dates.
+ */
+function keystoneDateToStatus(date: Temporal.PlainDate | null): {
+  baseline: BaselineStatus;
+  baseline_low_date: BaselineDate;
+  baseline_high_date: BaselineDate;
+} {
```

Comment:
A brand new method to find the status, which sidesteps the Set difference stuff.